File: problems/2391_minimum_amount_of_time_to_collect_garbage/solution.py
from typing import List


class Solution:
    def garbageCollection(self, garbage: List[str], travel: List[int]) -> int:
        result = 0
        last = {'M': 0, 'P': 0, 'G': 0}
        for i, g in enumerate(garbage):
            result += len(g)
            for c in g:
                last[c] = i
        prefix = list(accumulate(travel, initial=0))
        for v in last.values():
            result += prefix[v]
        return result

File: problems/2391_minimum_amount_of_time_to_collect_garbage/Solution.java
class Solution {
    public int garbageCollection(String[] garbage, int[] travel) {
        int result = 0;
        int[] last = new int[3];
        for (int i = 0; i < garbage.length; i++) {
            result += garbage[i].length();
            for (char c : garbage[i].toCharArray()) {
                if (c == 'M') last[0] = i;
                else if (c == 'P') last[1] = i;
                else last[2] = i;
            }
        }
        int[] prefix = new int[travel.length + 1];
        for (int i = 0; i < travel.length; i++) prefix[i+1] = prefix[i] + travel[i];
        for (int l : last) result += prefix[l];
        return result;
    }
}
